integreat_cms/cms/forms/contacts/contact_form.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from django import forms
+from django.utils.translation import gettext_lazy as _
+
+from ...models import Contact
+from ..custom_model_form import CustomModelForm
+
+if TYPE_CHECKING:
+    from typing import Any
+
+logger = logging.getLogger(__name__)
+
+
+class ContactForm(CustomModelForm):
+    """
+    Form for creating and modifying contact objects
+    """
+
+    class Meta:
+        """
+        This class contains additional meta configuration of the form class, see the :class:`django.forms.ModelForm`
+        for more information.
+        """
+
+        #: The model of this :class:`django.forms.ModelForm`
+        model = Contact
+        #: The fields of the model which should be handled by this form
+        fields = [
+            "title",
+            "name",
+            "location",
+            "email",
+            "phone_number",
+            "website",
+        ]
+
+        error_messages = {
+            "location": {"invalid_choice": _("Location cannot be empty.")}
+        }
+
+    def clean(self) -> dict[str, Any]:
+        """
+        Validate the fields
+
+        :return: The cleaned form data
+        """
+        cleaned_data = super().clean()
+
+        if (
+            not self.data["title"]
+            and not self.data["name"]
+            and not self.data["email"]
+            and not self.data["phone_number"]
+            and not self.data["website"]
+        ):
+            self.add_error(
+                None,
+                forms.ValidationError(
+                    _(
+                        "One of the following fields must be filled: title, name, E-mail, phone number, website."
+                    ),
+                    code="invalid",
+                ),
+            )
+
+        related_location = self.instance.region.pois.filter(
+            id=self.data["location"]
+        ).first()
+
+        contacts_with_empty_title = Contact.objects.filter(
+            title="", location=related_location
+        )
+        if self.instance.id:
+            contacts_with_empty_title = contacts_with_empty_title.exclude(
+                id=self.instance.id
+            )
+
+        if not self.data["title"] and contacts_with_empty_title:
+            self.add_error(
+                None,
+                forms.ValidationError(
+                    _("Only one contact per location can have empty title."),
+                    code="invalid",
+                ),
+            )
+
+        return cleaned_data
```

Comment:
```suggestion
```

The problem with doing the validation here, is that this code is potentially prone to race-conditions (If two forms are submitted at similar times, then for both forms this check could run simultaneously, see no errors, and write to the database). Instead, I'd propose to delegate the task of checking these conditions to the database, by adding two constraints to the contact model:

```python
        constraints = [
            models.UniqueConstraint("location", condition=Q(title=""), name = "contact_singular_empty_title_per_location", violation_error_message=_("Only one contact per location can have an empty title.")),
            models.CheckConstraint(check=~Q(title="")|~Q(name="")|~Q(email="")|~Q(phone_number="")|~Q(website=""), name="contact_non_empty", violation_error_message=_("One of the following fields must be filled: title, name, e-mail, phone number, website."))
        ]
```